examples/keras_recipes/tensorflow_numpy_models.py

```diff
@@ -0,0 +1,340 @@
+descent is handled automatically when using the `tnp` API alongside keras.
+"""
+
+"""
+First let's define a simple `TNPForwardFeedRegressionNetwork` class.
+"""
+
+
+class TNPForwardFeedRegressionNetwork(keras.models.Model):
+    def __init__(self, blocks=None):
+        super(TNPForwardFeedRegressionNetwork, self).__init__()
+        if not isinstance(blocks, list):
+            raise ValueError(f"blocks must be a list, got={blocks}")
+        self.blocks = blocks
+        self.block_weights = None
+        self.biases = None
+
+    def build(self, input_shape):
+        current_shape = input_shape[1]
+        self.block_weights = []
+        self.biases = []
+        for i, block in enumerate(self.blocks):
+            self.block_weights.append(
+                self.add_weight(
+                    shape=(current_shape, block), trainable=True, name=f"block-{i}"
+                )
+            )
+            self.biases.append(
+                self.add_weight(shape=(block,), trainable=True, name=f"bias-{i}")
+            )
+            current_shape = block
+
+        self.linear_layer = self.add_weight(
+            shape=(current_shape, 1), name="linear_projector", trainable=True
+        )
+
+    def call(self, inputs):
+        activations = inputs
+        for w, b in zip(self.block_weights, self.biases):
+            activations = tnp.matmul(activations, w) + b
+            # ReLu activation function
+            activations = tnp.maximum(activations, 0)
+
+        return tnp.matmul(activations, self.linear_layer)
+
+
+"""
+Just like with any other Keras model we can utilize any supported optimizer, loss,
+metrics or callbacks that we want.
+"""
+
+"""
+Let's see how the model performs!
+"""
+
+model = TNPForwardFeedRegressionNetwork(blocks=[3, 3])
+model.compile(
+    optimizer="adam",
+    loss="mean_squared_error",
+    metrics=[keras.metrics.MeanAbsolutePercentageError()],
+)
+evaluate_model(model)
+
+"""
+Great!  Our model seems to be effectively learning to solve the problem at hand.
+"""
+
+"""
+# Implementing a Keras Layer Based Model with TNP
+If desired, `tnp` can also be used in layer oriented Keras code structure.  Let's
+implement the same model, but using a layered approach!
+"""
+
+
+class DenseTNP(keras.layers.Layer):
+    def __init__(self, units):
+        super().__init__()
+        self.units = units
+
+    def build(self, input_shape):
+        self.w = self.add_weight(
+            name="weights",
+            shape=(input_shape[1], self.units),
+            initializer="random_normal",
+            trainable=True,
+        )
+        self.bias = self.add_weight(
+            name="bias",
+            shape=(self.units,),
+            initializer="random_normal",
+            trainable=True,
+        )
+
+    def call(self, inputs):
+        return tnp.maximum(tnp.matmul(inputs, self.w) + self.bias, 0)
+
+
+class LinearProjector(keras.layers.Layer):
```

Comment:
Once you make `activation` modular, this is just a default dense layer with 1 unit.

Reply:
Done